src/vs/workbench/contrib/preferences/browser/settingsTree.ts

```diff
@@ -1324,39 +1350,51 @@ export class SettingTextRenderer extends AbstractSettingRenderer implements ITre
 
 		renderValidations(dataElement, template, true);
 	}
+}
 
-	private createInputBox(dataElement: SettingsTreeSettingElement, template: ISettingTextItemTemplate): InputBox {
-		const useMultiline = dataElement.setting.editPresentation === 'multilineText';
-		const inputBoxOptions: IInputOptions = {
-			flexibleHeight: useMultiline,
-			flexibleWidth: false
-		};
-		const inputBox = new InputBox(template.controlElement, this._contextViewService, inputBoxOptions);
-		template.toDispose.add(inputBox);
-		template.toDispose.add(attachInputBoxStyler(inputBox, this._themeService, {
-			inputBackground: settingsTextInputBackground,
-			inputForeground: settingsTextInputForeground,
-			inputBorder: settingsTextInputBorder
+export class SettingTextRenderer extends AbstractSettingTextRenderer implements ITreeRenderer<SettingsTreeSettingElement, never, ISettingTextItemTemplate> {
+	templateId = SETTINGS_TEXT_TEMPLATE_ID;
+
+	override renderTemplate(_container: HTMLElement): ISettingTextItemTemplate {
+		super.useMultiline = false;
+		const template = super.renderTemplate(_container);
+
+		// TODO@9at8: listWidget filters out all key events from input boxes, so we need to come up with a better way
+		// Disable ArrowUp and ArrowDown behaviour in favor of list navigation
+		template.toDispose.add(DOM.addStandardDisposableListener(template.inputBox.inputElement, DOM.EventType.KEY_DOWN, e => {
+			if (e.equals(KeyCode.UpArrow) || e.equals(KeyCode.DownArrow)) {
+				e.preventDefault();
+			}
 		}));
+
+		return template;
+	}
+}
+
+export class SettingMultilineTextRenderer extends AbstractSettingTextRenderer implements ITreeRenderer<SettingsTreeSettingElement, never, ISettingTextItemTemplate> {
+	templateId = SETTINGS_MULTILINE_TEXT_TEMPLATE_ID;
+
+	override renderTemplate(_container: HTMLElement): ISettingTextItemTemplate {
+		super.useMultiline = true;
```

Comment:
I think it would be better to either set this in the constructor, or pass it to super.renderTemplate as a parameter

Reply:
I'll pass it as a parameter